src/ManagedServices/ManagedServices/Commands/GetAzureRmManagedServicesAssignment.cs

```diff
@@ -0,0 +1,79 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Commands
+{
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet(
+        VerbsCommon.Get,
+        Microsoft.Azure.Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ManagedServicesAssignment",
+        DefaultParameterSetName = DefaultParameterSet,
+        SupportsShouldProcess = true), OutputType(typeof(PSRegistrationAssignment))]
+    public class GetAzureRmManagedServicesAssignment : ManagedServicesCmdletBase
+    {
+        protected const string DefaultParameterSet = "Default";
+
+        [Parameter(Position = 0, ParameterSetName = DefaultParameterSet, HelpMessage = "The subscription scope.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = DefaultParameterSet, ValueFromPipeline = true, HelpMessage = "The Registration Assignment Id.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("Id")]
+        public string ResourceId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, HelpMessage = "Whether to include the Registration Definiton in the response.")]
+        public SwitchParameter ExpandRegistrationDefinition { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var assignmentId = this.ResourceId.IsGuid() ? this.ResourceId : this.ResourceId?.GetResourceName();
+            this.TryCatchExecuteCmdlet(() =>
+            {
+                ConfirmAction(MyInvocation.InvocationName,
+                    "Microsoft.ManagedServices/registrationAssignments",
+                    () =>
+                    {
+                        var scope = this.Scope ?? this.GetDefaultScope();
+
+                        if (string.IsNullOrEmpty(assignmentId))
+                        {
+                            var results = this.PSManagedServicesClient.ListRegistrationAssignments(
+                                scope: scope,
+                                expandRegistrationDefinition: this.ExpandRegistrationDefinition);
+                            this.WriteRegistrationAssignmentList(results);
+                        }
+                        else
+                        {
+                            // validate assignmentId.
+                            if (!assignmentId.IsGuid())
+                            {
+                                throw new ApplicationException("RegistrationAssignment must be a valid GUID.");
+                            }
+
+                            var result = this.PSManagedServicesClient.GetRegistrationAssignment(
```

Comment:
Thsi is not the intended usage of the resource id parameter set - the idea is to be able to pass in a resourceId for the assignment, without having to separately pass in a scope.